### PR TITLE
feat: add support for bulk operations for users and permission groups

### PR DESCRIPTION
**Pull request by eshankvaish**
# Description

- Add bulk delete and bulk status updates for users
- Add bulk delete for permission groups

Fixes https://github.com/devtron-labs/devtron/issues/4640

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/css/formulae.scss [hunk 1]
```diff
@@ -116,6 +116,11 @@ $style in $margins {
     .fs-#{$i} {
         font-size: 0px + $i;
     }
+
+    // To override the custom font-size
+    .fs-#{$i}-imp {
```
Comment (vivek-devtron) [inline src/css/formulae.scss:121]: We should avoid doing this unless we require this extensively.

File: src/Pages/GlobalConfigurations/Authorization/UserPermissions/List/UserPermissionContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    bulkSelectionModalConfig,
+    setBulkSelectionModalConfig,
+}: UserPermissionContainerProps) => {
+    const isClearBulkSelectionModalOpen = !!bulkSelectionModalConfig?.type
+
+    const { searchKey, handleSearch: _handleSearch, clearFilters, status, updateStatusFilter } = urlFilters
+
+    const draggableRef = useRef<HTMLDivElement>()
+    const { getSelectedIdentifiersCount, isBulkSelectionApplied } = useAuthorizationBulkSelection()
+    const isSomeRowChecked = getSelectedIdentifiersCount() > 0
```
Comment (vivek-devtron) [inline src/Pages/GlobalConfigurations/Authorization/UserPermissions/List/UserPermissionContainer.tsx:42]: Please check if we can avoid this

File: src/Pages/GlobalConfigurations/Authorization/UserPermissions/List/UserPermissionList.component.tsx [hunk 3]
```diff
@@ -69,118 +92,38 @@ const UserPermissionList = () => {
...
+        >
+            <UserPermissionContainer
                 showStatus={showStatus}
-                handleSearch={handleSearch}
-                initialSearchText={searchKey}
-                getDataToExport={getUserDataForExport}
+                error={error}
+                getUserDataForExport={getUserDataForExport}
+                showLoadingState={showLoadingState}
+                totalCount={result?.totalCount}
```
Comment (vivek-devtron) [inline src/Pages/GlobalConfigurations/Authorization/UserPermissions/List/UserPermissionList.component.tsx:119]: Should we default it to 0

File: src/Pages/GlobalConfigurations/Authorization/UserPermissions/List/UserPermissionRow.tsx [hunk 4]
```diff
@@ -33,11 +37,17 @@ const UserPermissionRow = ({
     index,
     showStatus,
     refetchUserPermissionList,
+    isChecked = false,
+    toggleChecked,
+    showCheckbox: _showCheckbox,
```
Comment (vivek-devtron) [inline src/Pages/GlobalConfigurations/Authorization/UserPermissions/List/UserPermissionRow.tsx:42]: We can make local variable with _ and let it be as is

File: src/Pages/GlobalConfigurations/Authorization/UserPermissions/List/UserPermissionContainer.tsx (hunk 2 above)
Comment (eshankvaish) [reply src/Pages/GlobalConfigurations/Authorization/UserPermissions/List/UserPermissionContainer.tsx:42]: Will retest and fix in upcoming PR

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**